zulipterminal/model.py

```diff
@@ -701,7 +707,19 @@ def _handle_message_event(self, event: Event) -> None:
                 self.controller.update_screen()
 
         # We can notify user regardless of whether UI is rendered or not.
```

Comment:
Now that we need the UI to have rendered to show notifications, should we remove this comment? Should we do some bounded waiting here?

Reply:
I've updated this comment accordingly, and changed the code to only render to the UI if it is ready. We could really do with an analysis of the code to understand if we can simplify this, as this kind of check exists elsewhere. I suspect it will be easier if the loading screen is in `urwid`, as the UI loads first then in any case.

Reply:
Thanks for the quick update. Yes, we should simplify this.